src/seedsigner/views/seed_views.py

```diff
@@ -2299,3 +2309,330 @@ def run(self):
 
         # Exiting/Canceling the QR display screen always returns Home
         return Destination(MainMenuView, skip_current_view=True)
+
+
+
+"""****************************************************************************
+    Shamir's Secret Sharing Views
+****************************************************************************"""
+class SeedShamirShareStartView(View):
+    """
+    Currently just a warning display before entering the Shamir shares.
+    
+    Could be expanded with a follow-up View to specify Shamir seed type.
+    """
+    def run(self):
+        self.run_screen(
+                WarningScreen,
+                title=_("Shamir Secret Sharing"),
+                status_headline=None,
+                text=_("Some features are disabled for Shamir seeds."),
+                show_back_button=False,
+        )
+
+        return Destination(SeedShamirShareImportSelectWordCount)
+
+
+
+class SeedShamirShareImportSelectWordCount(View):
+    TYPE_20WORD = ButtonOption("20 words")
+    TYPE_33WORD = ButtonOption("33 words")
+
+    def __init__(self):
+        super().__init__()
+
+
+    def run(self):
+        button_data = [self.TYPE_20WORD, self.TYPE_33WORD]
+
+        selected_menu_num = self.run_screen(
+            ButtonListScreen,
+            title=_("Words Per Share"),
+            button_data=button_data,
+        )
+        
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(MainMenuView, clear_history=True)
+
+        elif button_data[selected_menu_num] == self.TYPE_20WORD:
+            self.controller.storage.init_pending_shamir_share_set(num_words=20)
+            return Destination(SeedShamirShareMnemonicEntryView)
+        
+        elif button_data[selected_menu_num] == self.TYPE_33WORD:
+            self.controller.storage.init_pending_shamir_share_set(num_words=33)
+            return Destination(SeedShamirShareMnemonicEntryView)
+
+
+
+class SeedShamirShareMnemonicEntryView(View):
+    def __init__(self, cur_word_index: int = 0):
+        super().__init__()
+        self.cur_word_index = cur_word_index
+
+        # Ensure pending mnemonic is initialized for the current share
+        if not self.controller.storage.pending_mnemonic or len(self.controller.storage.pending_mnemonic) == 0:
+            # Initialize pending mnemonic with the correct word count
+            num_words = self.controller.storage.pending_shamir_num_words
+            self.controller.storage.init_pending_mnemonic(num_words)
+        
+        self.cur_word = self.controller.storage.get_pending_mnemonic_word(cur_word_index)
+
+
+    def run(self):
+        share_num = self.controller.storage.pending_shamir_share_set_length + 1
+        
+        ret = self.run_screen(
+            seed_screens.SeedMnemonicEntryScreen,
+            title=_("Share#{}, Word#{}").format(share_num, self.cur_word_index+1),  # Using spaces here "Share #{}, Word #{}" makes this screen laggish
```

Comment:
I think it looks odd without the spaces.

If you want to avoid scrolling in the title, I'd just drop the "Share" reference entirely. If the user loses their place and can't remember which share num they're entering... um... no pity here.

The other note is that if a translated phrase has two variable insertions, it's a best practice to use named variables in the curly braces (it's unlikely to be an issue in this case, but for some translations we can't guarantee that the order will always be the same: "I have {currency_symbol}{fiat_amount}" vs "Tengo {fiat_amount}{currency_symbol}" for a locale that prefers "5€" vs "$5". Named variables give the translators the freedom to insert each one in its proper place).